pkg/orchestrator/orchestrator.go

```diff
@@ -56,14 +56,17 @@ type OrchestratorSettings struct {
 	JenkinsUser  string
 	JenkinsToken string
 	AzureToken   string
+	GitHubToken  string
 }
 
 func NewOrchestratorSpecificConfigProvider() (OrchestratorSpecificConfigProviding, error) {
 	switch DetectOrchestrator() {
 	case AzureDevOps:
 		return &AzureDevOpsConfigProvider{}, nil
 	case GitHubActions:
-		return &GitHubActionsConfigProvider{}, nil
+		return gitHubActionsConfigProvider(&OrchestratorSettings{
+			GitHubToken: getEnv("PIPER_ACTION_GITHUB_TOOLS_TOKEN", ""),
+		})
```

Comment:
It's a bit different from how it works with other orchestrators, and also in this way we use an env var to get the token, as opposed to the config (Vault) for ADO. Which I think is fine because then we only have to keep it in one place. But either slight refactoring or comments explaning the inconsistency with ADO are needed in my opinion. Let's discuss further on Monday.

Reply:
After a discussion with the team, we decided that implementation should be consistent with other orchestrators. Changes were pushed and currently, it is consistent with ADO.